README.md

```diff
@@ -467,21 +467,78 @@ That also covers gdb-based GUI - e.g. Visual Studio Code `Debug`.
 Payload running as a unikernel in Kontain VM will generate a coredump in the same cases it would have generated if running on Linux. The file name is `kmcore` (can be changed with `--coredump=file` flag).
 You can analyze the payload coredump as a regular Linux coredump, e.g. `gdb program.km kmcore`
 
-### Live debugging - command line
+### Live payload debugging - command line
 
-Pass `-g` flag to km to start in "GDB" mode, and it will print `gdb` command to run to start debugging session right away.
- e.g. :
+You can attach the standard gdb client to a km payload if you arrange for the gdb server inside km to be listening for a connection from the gdb client.
+Several flags control km's activation of the internal gdb server.
+The simplest flag "-g[port]" tells km to start the gdb server listening on the specified port and then wait for a connection from
+the gdb client.  The default gdb listening port is 2159 (gdbremote from /etc/services).  The km gdb server only uses tcp.
+When you attach to the gdb server the payload is waiting to begin running at the _start entry point in the km payload.
+
+You can also have km's gdb server listening for connections in the background by using the km \-\-gdb-listen command line flag.
+With this flag the payload begins running immediately and the gdb server listens for gdb client connection while the payload runs.
+You can connect to the gdb server, disconnect, and reconnect as often as you want until the payload completes.
+When you connect to the km gdb server all payload threads will be paused until the gdb client starts them using the "cont" or "step" or "next" commands.
+
+You can only connect to a payload's gdb server from the machine hosting the payload.
+
+km gdb server testing has been done using gdb client with version "GNU gdb (GDB) Fedora 9.1-5.fc32".
+
+The km implementation uses one payload signal to pause payload threads while the gdb client is stopping the payload.  That is signal 63.
+To avoid seeing gdb client messages about this signal it is a good idea to either manually run the "handle SIG63 nostop" command for each debugging
+session or add this command to your ~/.gdbinit file.
+
+If your payload uses fork then the child payload resets all of the gdb related settings so the child payload can't be debugged.  We are working on correcting this shortfall.
+When the child payload finally exec's to a new payload, no gdb related arguments are added to the invocation of km.
+So, there is no debugging in the exec'ed payload either.  We are working on this too.
+
+#### Known gdb related issues
+
+Stack traces are sometimes incomplete.
+gdb non-stop mode (set non-stop on) is not supported.
+Do not alter the GS segment selector in your programs.  It is used internally by km.
```

Comment:
this does not have to be here. At the very least, it should be separate, i.e. 

This does not work in GDB: 
- set non-stop on (btw, what happens if it's typed? ) 
- info registers does not report floating point

Other know issues, FYI :
- GS is used internally, do not alter it in your code 
- dynamically linked payload consumes one FD per lib

Reply:
Here is what happens if you try to change non-stop:
```
(gdb) show non-stop
Controlling the inferior in non-stop mode is off.
(gdb) set non-stop on
Cannot change this setting while the inferior is running.
(gdb) set non-stop off
Cannot change this setting while the inferior is running.
(gdb)
```

Reply:
good. I think just adding "Gdb will respond with `Cannot change this setting while the inferior is running.`" to the bullen point above

Reply:
Removed